src/test/run-pass/union/union-manuallydrop.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// run-pass
+
+use std::mem::needs_drop;
+use std::mem::ManuallyDrop;
+
+struct NeedDrop;
+
+impl Drop for NeedDrop {
+    fn drop(&mut self) {}
+}
+
+#[allow(dead_code)]
+union UnionOk1<T> {
+    empty: (),
+    value: ManuallyDrop<T>,
+}
+
+#[allow(dead_code)]
+union UnionOk2 {
+    value: ManuallyDrop<NeedDrop>,
+}
+
+#[allow(dead_code)]
+union UnionOk3<T: Copy> {
```

Comment:
It would also be good to add a test that checks how the compiler handles:
```rust
trait ImpliesCopy: Copy {}

union UnionOk4<T: ImpliesCopy> {
    empty: (),
    value: T,
}
```

A test file checking that `T :/ Copy` will lead to an error would also be good.

Reply:
The ui tests touched in the PR cover the non-Copy case, but do we need another test?

Reply:
Ok, can't have a failing case in run-pass. It looks like this is normally in ui tests now, so I can add it there (I can't see a single compile-fail test for unions).

Reply:
UI tests are afaik compile-fail by default; but yeah, `U5` needs to be added to a different file which has `// compile-fail`.